### PR TITLE
Use pnpm instead of Yarn

### PR DESCRIPTION
**Pull request by Josehower**
Use pnpm instead of Yarn throughout the repo

### CODE REVIEW

File: .github/workflows/export-notion-blocks-and-commit.yml [hunk 1]
```diff
@@ -15,15 +15,16 @@ jobs:
     timeout-minutes: 35
     steps:
       - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
```
Comment (Josehower) [inline .github/workflows/export-notion-blocks-and-commit.yml:18]: get pnpm action from registry

File: .github/workflows/export-notion-blocks-and-commit.yml [hunk 2]
```diff
@@ -15,15 +15,16 @@ jobs:
...
+        with:
+          version: '7'
```
Comment (Josehower) [inline .github/workflows/export-notion-blocks-and-commit.yml:20]: use version 7 taken from example https://github.com/karlhorky/archive-webpage-browser-extension/blob/main/.github/workflows/lint-check-types-and-build.yml

File: .github/workflows/export-notion-blocks-and-commit.yml [hunk 3]
```diff
@@ -15,15 +15,16 @@ jobs:
...
       - uses: actions/setup-node@v3
         with:
           node-version: '18'
-
-      - name: Install deps and build (with cache)
-        uses: bahmutov/npm-install@v1
+          cache: 'pnpm'
```
Comment (Josehower) [inline .github/workflows/export-notion-blocks-and-commit.yml:24]: cache from pnpm taken from example https://github.com/karlhorky/archive-webpage-browser-extension/blob/main/.github/workflows/lint-check-types-and-build.yml

File: .github/workflows/export-notion-blocks-and-commit.yml [hunk 4]
```diff
@@ -15,15 +15,16 @@ jobs:
...
 
       - name: Run backup script
-        run: yarn tsx index.ts
+        run: pnpm tsx index.ts
```
Comment (Josehower) [inline .github/workflows/export-notion-blocks-and-commit.yml:27]: use pnpm instead of yarn

File: .github/workflows/lint-and-check-types.yml [hunk 5]
```diff
@@ -7,10 +7,15 @@ jobs:
...
+          version: '7'
       - uses: actions/setup-node@v3
         with:
           node-version: '18'
+          cache: 'pnpm'
       - name: Install deps and build (with cache)
-        uses: bahmutov/npm-install@v1
-      - run: yarn eslint . --max-warnings 0
-      - run: yarn tsc
+        run: pnpm install
```
Comment (Josehower) [inline .github/workflows/lint-and-check-types.yml:18]: install dependencies

File: .github/workflows/lint-and-check-types.yml [hunk 6]
```diff
@@ -7,10 +7,15 @@ jobs:
...
+      - run: pnpm eslint . --max-warnings 0
+      - run: pnpm tsc
```
Comment (Josehower) [inline .github/workflows/lint-and-check-types.yml:20]: compiles using tsc

File: .github/workflows/lint-and-check-types.yml [hunk 7]
```diff
@@ -7,10 +7,15 @@ jobs:
...
+      - run: pnpm tsc --project tsconfig.json --noEmit
```
Comment (Josehower) [inline .github/workflows/lint-and-check-types.yml:21]: compiles based on the tsconfig file taken from example
https://github.com/karlhorky/archive-webpage-browser-extension/blob/main/.github/workflows/lint-check-types-and-build.yml

**Review by karlhorky — APPROVED** [review]

**Pull request merged**